packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -130,23 +135,25 @@ export class MirrorNodeClient {
         return axiosClient;
     }
 
-    constructor(baseUrl: string, logger: Logger, register: Registry, axiosClient?: AxiosInstance) {
-        if (axiosClient !== undefined) {
-            this.baseUrl = '';
-            this.client = axiosClient;
-        } else {
-            if (!baseUrl.match(/^https?:\/\//)) {
-                baseUrl = `https://${baseUrl}`;
-            }
+    constructor(restUrl: string, logger: Logger, register: Registry, restClient?: AxiosInstance, web3Url?: string, web3Client?: AxiosInstance) {
+        if (!web3Url) {
+            web3Url = restUrl;
+        }
 
-            if (!baseUrl.match(/\/$/)) {
-                baseUrl = `${baseUrl}/`;
-            }
+        if (restClient !== undefined) {
+            this.restUrl = '';
+            this.web3Url = '';
 
-            baseUrl = `${baseUrl}api/v1/`;
+            this.restClient = restClient;
+            this.web3Client = !!web3Client ? web3Client : restClient;
+        } else {
+            restUrl = this.buildUrl(restUrl);
```

Comment:
I forget why we try to update the param passed in.
Why not
```suggestion
            this.restUrl = this.buildUrl(restUrl);
```